File: pyg-cart/pyg-cart-interface/src/main/java/com/pyg/cart/service/CartService.java
package com.pyg.cart.service;

import com.pyg.vo.Cart;

import java.util.List;

/**
 * Created by on 2018/8/26.
 */
public interface CartService {

    /**
     * 需求：查询redis购物车
     * 参数：String userId
     * 返回值:List<cart>
     */
    public List<Cart> findRedisCartList(String userId);
    /**
     * 需求：组装购物车数据结构
     * 参数：
     * 参数1：List<Cart> cartList  原来的购物车列表
     * 参数2：商品id
     * 参数3：购买的商品数量 num
     * 返回值：List<Cart>
     */
    public List<Cart> addCartList(List<Cart> cartList,Long itemId,Integer num);

    /**
     * 需求：添加redis购物车
     * @param cartList
     * @param username
     */
    public void addRedisCart(List<Cart> cartList, String username);

    /**
     * 需求：合并购物车数据
     * @param redisCartList
     * @param cookieCartList
     * @return
     */
    List<Cart> mergeCart(List<Cart> redisCartList, List<Cart> cookieCartList);
}

File: pyg-cart-web/src/main/webapp/js/service/cartService.js
//服务层
app.service('cartService', function ($http) {

    //查询购物车数据
    this.findCartList = function () {
        return $http.get("../cart/findCartList");
    }

    //加1，减一购物车数量
    this.addGoodsToCartList = function (itemId, num) {
        return $http.get("../cart/addGoodsToCartList/" + itemId + "/" + num);
    };

    //计算购物车列表中总价格，总数量
    this.sum = function (cartList) {

        //定义对象，封装数量和价格
        var totalValue = {totalNum: 0, totalPrice: 0};

        for (var i = 0; i < cartList.length; i++) {
            //获取购物车清单数据
            var orderItemList = cartList[i].orderItemList;
            //循环购物车列表集合
            for (var j = 0; j < orderItemList.length; j++) {
                totalValue.totalNum += orderItemList[j].num;
                totalValue.totalPrice += orderItemList[j].totalFee;
            }
        }

        return totalValue;


    }


});
